users/password-change.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/** 
+ * Postfix Admin 
+ * 
+ * LICENSE 
+ * This source file is subject to the GPL license that is bundled with  
+ * this package in the file LICENSE.TXT. 
+ * 
+ * Further details on the project are available at http://postfixadmin.sf.net 
+ * 
+ * @version $Id$ 
+ * @license GNU GPL v2 or later. 
+ * 
+ * File: password-change.php
+ * Used by users and admins to change their forgotten login password.
+ * Template File: password-change.tpl
+ *
+ * Template Variables:
+ *
+ * tUsername
+ * tCode
+ *
+ * Form POST \ GET Variables:
+ *
+ * fUsername
+ */
+
+if (preg_match('/\/users\//', $_SERVER['REQUEST_URI'])) {
+  $rel_path = '../';
+  $context = 'users';
+} else {
+  $rel_path = './';
+  $context = 'admin';
+}
+require_once($rel_path . 'common.php');
+
+if ($context == 'admin' && !Config::read('forgotten_admin_password_reset') || $context == 'users' && !Config::read('forgotten_user_password_reset'))
+{
+    header('HTTP/1.0 403 Forbidden');
+    exit(0);
+}
+
+if ($_SERVER['REQUEST_METHOD'] == 'GET')
+{
+    $tUsername = safeget('username');
+    $tCode = safeget('code');
+}
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST')
+{
+    if(safepost('fCancel')) {
+        header('Location: main.php');
+        exit(0);
+    }
+
+    $fPassword = safepost('fPassword');
+    $fPassword2 = safepost('fPassword2');
+
+    $tUsername = safepost('fUsername');
+    $tCode = trim(strtoupper(safepost('fCode')));
+
+    if (empty($fPassword) or ($fPassword != $fPassword2)) {
+        $error = true;
+        flash_error(Config::lang('pPassword_password_text_error'));
+    } elseif (trim(strtoupper($tCode) != getPasswordRecoveryCode($tUsername))) {
+        flash_error(Config::lang('pPassword_code_text_error'));
+    } else {
+        session_regenerate_id();
+        $_SESSION['sessid']['username'] = $tUsername;
+        if ($context == 'users') {
```

Comment:
If the login code can be refactored in a shared function, then I would vote in favor of auto-login after password reset.

I don't think it could be a security concern, since the user was just authorized to change his password anyway. And removing a step is indeed convenient for the user, because it's extremely likely that what he wanted to do in the first place was to login.